ikarus/controlroutines/controlroutinebase.hh

```diff
@@ -0,0 +1,22 @@
+// SPDX-FileCopyrightText: 2021-2025 The Ikarus Developers mueller@ibb.uni-stuttgart.de
+// SPDX-License-Identifier: LGPL-3.0-or-later
+
+/**
+ * \file controlroutinebase.hh
+ * \brief Base for all control routines
+ */
+
+#pragma once
+#include <ikarus/utils/broadcaster/broadcaster.hh>
+#include <ikarus/utils/broadcaster/broadcastermessages.hh>
+
+namespace Ikarus {
+
+struct ControlRoutineBase
+    : public Broadcasters<void(ControlMessages), void(ControlMessages, const std::string&),
+                          void(ControlMessages, int, const std::string&), void(ControlMessages, int, double)>
+
+{
```

Comment:
Same here then, we also want a broadcaster with a controlroutine state. Then this base have to be templated though...